src/tiled/documentmanager.cpp

```diff
@@ -968,3 +1028,31 @@ bool DocumentManager::askForAdjustment(const Tileset &tileset)
 
     return r == QMessageBox::Yes;
 }
+
+bool DocumentManager::eventFilter(QObject *, QEvent *event)
+{
+    switch (event->type()) {
+        case QEvent::MouseButtonRelease: {
+            // middle-click tab closing
+            QMouseEvent *mevent = static_cast<QMouseEvent*>(event);
+
+            if (mevent->button() == Qt::MidButton) {
+                int index = mTabBar->tabAt(mevent->pos());
+
+                if (index != -1) {
+                    documentCloseRequested(index);
+                    return true;
+                }
+            }
+        }
+        break;
+    default:
+        break;
+    }
+
+    return false;
+}
+
+void DocumentManager::abortMultiDocumentClose() {
```

Comment:
Coding style: for function definitions the opening `{` should be on its own line.